SwiftCompilerSources/Sources/Optimizer/PassManager/Options.swift

```diff
@@ -68,4 +68,8 @@ struct Options {
       default:                   return .unknown
     }
   }
+
+  var enablePerformanceDiagnostics: Bool {
```

Comment:
This is a bad name. Originally, `SILOptions::EnablePerformanceDiagnostics` was only used to disable the PerformanceDiagnosticsPass pass. But in embedded swift, such diagnostics are not "performance" related at all.
Can we just call it `isInSourceKit` or something like that?
And add a comment that in SourceKit we can't emit certain diagnostics which rely on wmo, which SourceKit doesn't support.

Also, as this is not a command line option, we should move this property directly into `Context` (at least on the swift side).

Reply:
How would you feel about `skipWMORequiredDiagnostics` or `enableWMORequestDiagnostics` or something similar? I'd rather make this more generic than sourcekit specifically, even if it is the only place we do this today.

Reply:
Sounds reasonable. Please add a comment that this is (currently) the case in SourceKit